### PR TITLE
feat: add parameter to specify acceptable status of build to restart

### PR DESCRIPTION
**Pull request by meinenec**
Allows users to specify what build statuses are acceptable to restart. The available options are defined in [go-vela/types](https://github.com/go-vela/types/blob/master/constants/status.go)

The default behavior remains unchanged, if no status parameter is provided the build will run the last `success` build.

Example:
```
steps:
  - name: downstream
    image: hub.docker.target.com/target/vela-downstream
    parameters:
      status:
        - success
        - canceled
      repos:
        - myorg/mydownstreamrepo
```
Will restart the most recent build on myorg/mydownstreamrepo with either a success of canceled status.

### CODE REVIEW

File: cmd/vela-downstream/main.go [hunk 1]
```diff
@@ -96,6 +97,13 @@ func main() {
 			Name:     "repo.names",
 			Usage:    "list of <org>/<repo> names to trigger",
 		},
+		&cli.StringSliceFlag{
+			EnvVars:  []string{"PARAMETER_STATUS", "DOWNSTREAM_STATUS"},
+			FilePath: "/vela/parameters/downstream/status,/vela/secrets/downstream/status",
+			Name:     "status",
+			Usage:    "status of last build to trigger - options: (error|failure|killed|canceled|pending|running|success)",
```
[the remark this answers is not in the document]
Comment (meinenec) [reply cmd/vela-downstream/main.go:104]: thinking on how to improve this line

File: cmd/vela-downstream/plugin.go [hunk 2]
```diff
@@ -79,12 +80,12 @@ func (p *Plugin) Exec() error {
 			opts.Page = resp.NextPage
 		}
 
-		logrus.Debugf("Searching for latest successful build with branch %s", repo.GetBranch())
+		logrus.Debugf("Searching for latest build on branch %s with status %s", repo.GetBranch(), p.Status)
```
[the remark this answers is not in the document]
Comment (meinenec) [reply cmd/vela-downstream/plugin.go:83]: thinking on how to improve this line

File: cmd/vela-downstream/plugin.go [hunk 3]
```diff
@@ -95,7 +96,7 @@ func (p *Plugin) Exec() error {
 
 		// check if we found a build to restart
 		if build.GetNumber() == 0 {
-			return fmt.Errorf("no successful build with branch %s found for %s", repo.GetBranch(), repo.GetFullName())
+			return fmt.Errorf("no build with status %s on branch %s found for %s", p.Status, repo.GetBranch(), repo.GetFullName())
```
[the remark this answers is not in the document]
Comment (meinenec) [reply cmd/vela-downstream/plugin.go:99]: thinking on how to improve this line

**Review by jbrockopp — DISMISSED** [review]
LGTM

**Review by jbrockopp — APPROVED** [review]
LGTM

**Review by wass3r — APPROVED** [review]

**Pull request merged**